### ISSUE TITLE
Is this project alive?

### ISSUE THREAD

**Issue by Maniues** [matrix-construct/construct#202]
As in the title

**Comment by jevolk** [comment]
Yes. I haven't had time to make a release here yet, but the staging area for the next release is https://github.com/jevolk/charybdis. I can only recommend this if you're interested in developing some recent room versions and features like authenticated media before making use of it.

If you want a homeserver for casual use right now try https://github.com/girlbossceo/conduwuit

**Comment by bqv** [comment]
I never thought I'd see the day you uh... "sold out"

**Comment by gardiol** [comment]
Now that Conduwuit is dead as well, i feel sad.

**Comment by jevolk** [comment]
> Now that Conduwuit is dead as well, i feel sad.

Don't worry, it's a beautiful day in the neighborhood: https://github.com/matrix-construct/tuwunel

**Comment by gardiol** [comment]
This is awesome...
I will migrate immediately as soon as there is a release :)

Love open source, and will support this project (tuwunnel)

**Comment by Maniues** [comment]
> > Now that Conduwuit is dead as well, i feel sad.
> 
> Don't worry, it's a beautiful day in the neighborhood: https://github.com/matrix-construct/tuwunel

Is it planned to continue this project (tuwunel) in the long term, or it may be dead in the future (almost like The Construct).

**Comment by jevolk** [comment]
> > > Now that Conduwuit is dead as well, i feel sad.
> > 
> > 
> > Don't worry, it's a beautiful day in the neighborhood: https://github.com/matrix-construct/tuwunel
> 
> Is it planned to continue this project (tuwunel) in the long term, or it may be dead in the future (almost like The Construct).

@Maniues Sorry it wasn't appropriate for me to respond when you posted this as I was in the midst of negotiating the sponsorship which ensures tuwunel's long-term development. Construct development has indeed stalled and that's primarily due to lack of funding. It requires a full-time effort to develop a Matrix server, to deal with typical issues, keep up with the spec, and support users, etc.

The market is biased toward a Rust server, and given the foundations laid by Ruma and Conduit it simply made more sense for me to pivot and learn Rust to meet that market. This doesn't mean Construct is dead and I have not archived it because it's still possible for some cunning hacker to show up and bring it up to speed. In the meantime I have it on my todo to make another release here with a warning message that it's not suitable for general use.

The last time I used it a couple months ago media was obviously broken, there is no room v11 support, element was extremely broken due to their own lazy-loading non-compliance (though other clients probably work), and new panics are cropping up from unexpected spec fields missing (which is wild because everything was specifically written robustly against this sort of thing).
